pkg/apis/azure/validation/infrastructure.go

```diff
@@ -114,17 +92,83 @@ func ValidateInfrastructureConfig(infra *apisazure.InfrastructureConfig, nodesCI
+		return append(allErrs, field.Invalid(vnetConfigPath, vnetConfig, "a vnet cidr or vnet name and resource group need to be specified"))
+	}
+
+	if isExternalVnetUsed(&vnetConfig) {
+		if vnetConfig.CIDR != nil {
+			allErrs = append(allErrs, field.Invalid(vnetConfigPath.Child("cidr"), vnetConfig, "specifying a cidr for an existing vnet is not possible"))
+		}
+
+		if resourceGroupConfig != nil && *vnetConfig.ResourceGroup == resourceGroupConfig.Name {
+			allErrs = append(allErrs, field.Invalid(vnetConfigPath.Child("resourceGroup"), *vnetConfig.ResourceGroup, "the vnet resource group must not be the same as the cluster resource group"))
+		}
+		return allErrs
+	}
+
+	// Validate no cidr config is specified at all.
+	if vnetConfig.CIDR == nil {
+		allErrs = append(allErrs, workers.ValidateSubset(nodes)...)
+		allErrs = append(allErrs, workers.ValidateNotSubset(pods, services)...)
+		return allErrs
+	}
+
+	vnetCIDR := cidrvalidation.NewCIDR(*vnetConfig.CIDR, vnetConfigPath.Child("cidr"))
+	allErrs = append(allErrs, vnetCIDR.ValidateParse()...)
+	allErrs = append(allErrs, vnetCIDR.ValidateSubset(nodes)...)
+	allErrs = append(allErrs, vnetCIDR.ValidateSubset(workers)...)
+	allErrs = append(allErrs, vnetCIDR.ValidateNotSubset(pods, services)...)
+	allErrs = append(allErrs, cidrvalidation.ValidateCIDRIsCanonical(vnetConfigPath.Child("cidr"), *vnetConfig.CIDR)...)
+
+	return allErrs
+}
+
 // ValidateInfrastructureConfigUpdate validates a InfrastructureConfig object.
 func ValidateInfrastructureConfigUpdate(oldConfig, newConfig *apisazure.InfrastructureConfig, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 
 	allErrs = append(allErrs, apivalidation.ValidateImmutableField(newConfig.ResourceGroup, oldConfig.ResourceGroup, fldPath.Child("resourceGroup"))...)
-	allErrs = append(allErrs, apivalidation.ValidateImmutableField(newConfig.Networks.VNet, oldConfig.Networks.VNet, fldPath.Child("networks").Child("vnet"))...)
 	allErrs = append(allErrs, apivalidation.ValidateImmutableField(newConfig.Networks.Workers, oldConfig.Networks.Workers, fldPath.Child("networks").Child("workers"))...)
+	allErrs = append(allErrs, validateVnetConfigUpdate(&oldConfig.Networks, &newConfig.Networks, fldPath.Child("networks"))...)
 
 	if oldConfig.Zoned && !newConfig.Zoned {
 		allErrs = append(allErrs, field.Forbidden(fldPath.Child("zoned"), "moving a zoned cluster to a non-zoned cluster is not allowed"))
 	}
 
 	return allErrs
 }
+
+func validateVnetConfigUpdate(oldNeworkConfig, newNetworkConfig *apisazure.NetworkConfig, networkConfigPath *field.Path) field.ErrorList {
+	var allErrs = field.ErrorList{}
+
+	if newNetworkConfig.VNet.CIDR == nil && newNetworkConfig.VNet.Name == nil && newNetworkConfig.VNet.ResourceGroup == nil {
+		return append(allErrs, field.Invalid(networkConfigPath.Child("vnet"), newNetworkConfig.VNet, "a vnet cidr or vnet name and resource group need to be specified"))
```

Comment:
Since this is not an old vs new comparison, wouldn't that be covered in the normal validation `validateVnetConfig` ?

Reply:
Hmm no. It is valid to create a shoot without passing a vnet config at all. Then it will be defaulted. But it shouldn't be valid to remove the vnet config afterwards.

Reply:
Since we apply a default value in the controller itself here: https://github.com/gardener/gardener-extension-provider-azure/blob/master/pkg/internal/infrastructure/terraform.go#L118-L119

I agree it does not make sense to add this update validation yet. We should consider to add this defaulting not in the controller. Will open an issue for that and remove the validation for now.